modules/apps/bookmarks/bookmarks-service/src/main/java/com/liferay/bookmarks/internal/search/BookmarksEntryModelIndexerWriterContributor.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.bookmarks.internal.search;
+
+import com.liferay.bookmarks.model.BookmarksEntry;
+import com.liferay.bookmarks.service.BookmarksEntryLocalService;
+import com.liferay.portal.kernel.dao.orm.Property;
+import com.liferay.portal.kernel.dao.orm.PropertyFactoryUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.search.batch.BatchIndexingActionable;
+import com.liferay.portal.search.batch.DynamicQueryBatchIndexingActionableFactory;
+import com.liferay.portal.search.spi.model.index.contributor.ModelIndexerWriterContributor;
+import com.liferay.portal.search.spi.model.index.contributor.helper.ModelIndexerWriterDocumentHelper;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luan Maoski
+ */
+@Component(
+	immediate = true,
+	property = "indexer.class.name=com.liferay.bookmarks.model.BookmarksEntry",
+	service = ModelIndexerWriterContributor.class
+)
+public class BookmarksEntryModelIndexerWriterContributor
+	implements ModelIndexerWriterContributor<BookmarksEntry> {
+
+	@Override
+	public void customize(
+		BatchIndexingActionable batchIndexingActionable,
+		ModelIndexerWriterDocumentHelper modelIndexerWriterDocumentHelper) {
+
+		batchIndexingActionable.setAddCriteriaMethod(
+			dynamicQuery -> {
+				Property statusProperty = PropertyFactoryUtil.forName("status");
+
+				Integer[] statuses = {
+					WorkflowConstants.STATUS_APPROVED,
+					WorkflowConstants.STATUS_IN_TRASH
+				};
+
+				dynamicQuery.add(statusProperty.in(statuses));
+			});
+
+		batchIndexingActionable.setPerformActionMethod(
+			(BookmarksEntry bookmarksEntry) -> {
+				batchIndexingActionable.addDocuments(
+					modelIndexerWriterDocumentHelper.getDocument(
+						bookmarksEntry));
+
+				bookmarksFolderBatchReindexer.reindex(
```

Comment:
Hi @brandizzi.

On `BookmarksEntryIndexer` we have this code:

` Property folderIdProperty = PropertyFactoryUtil.forName(
	"folderId");
     dynamicQuery.add(folderIdProperty.eq(folderId)); `

We created the `BookmarksFolderBatchReindexer` to call a reindex passing the `folderId` as a parameter, but we are not sure if this is the correct behavior.

Is there a away to get the `folderId` inside the `setAddCriteriaMethod` of this class?

Reply:
@luanmaoski I read the code of `DefaultIndexer` and `IndexerWriterImpl` here and apparently you're right. I guess you'll have to implement a `ModelIndexerWriterContributor` for bookmarks with a `getBatchIndexingActionable()` method. Do you foresee any issues? If so, let me know!

Reply:
Alright. Thanks.